fix(linter): Fix the `react/jsx-fragments` rule config to take a string argument

connorshea opened the pull request:
The [original rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md) has a config option in the format of `["error", "element"]` or `["error", "syntax"]`. We required a config object like `["error", { "mode": "element" }]`, which is technically incompatible for existing configs that get migrated to oxlint.

This fixes that in the generated docs, and also makes it work correctly in the oxlint config. We retain compatibility with the object syntax in case anyone has set up oxlint with it already, and I've added tests to ensure both formats of the config option work.

Generated docs now:

```md
## Configuration

This rule accepts one of the following string values:

### `"syntax"`

This is the default mode. It will enforce the shorthand syntax for React fragments, with one exception.
Keys or attributes are not supported by the shorthand syntax, so the rule will not warn on standard-form fragments that use those.

Examples of **incorrect** code for this rule:
\```jsx
<React.Fragment><Foo /></React.Fragment>
\```

Examples of **correct** code for this rule:
\```jsx
<><Foo /></>
\```

\```jsx
<React.Fragment key="key"><Foo /></React.Fragment>
\```

### `"element"`

This mode enforces the standard form for React fragments.

Examples of **incorrect** code for this rule:
\```jsx
<><Foo /></>
\```

Examples of **correct** code for this rule:
\```jsx
<React.Fragment><Foo /></React.Fragment>
\```

\```jsx
<React.Fragment key="key"><Foo /></React.Fragment>
\```
```

PeterCardenas commented on crates/oxc_linter/src/rules/react/jsx_fragments.rs:
```diff
@@ -10,23 +10,26 @@ use crate::{AstNode, context::LintContext, rule::Rule, utils::is_jsx_fragment};
-    /// `syntax` mode:
-    ///
+    mode: FragmentMode,
+}
```
you can support both by doing an enum, something like this i think
```rs
#[schemars(untagged)]
pub enum JsxFragments {
  StringFragmentMode(FragmentMode);
  ObjectFragmentMode({ mode: FragmentMode });
}
```

connorshea replied:
I've pushed up a commit to use this pattern, although I'm not really sure it's particularly better than the previous version. I'll probably let cam decide on whatever is preferred.

PeterCardenas replied:
imo it's better because it codifies the config in the type rather than some adhoc deserialization code

connorshea replied:
Unfortunately we'll need to make it possible to serialize a value of this type for the website to generate it, right now it can't handle an enum that is more than just simple types.

connorshea replied:
DefaultRuleConfig definitely makes it better: 65f83444aa3e908e5464dbd27ce53380e24c0a40

connorshea committed:
fix(linter): Fix the `react/jsx-fragments` rule config to take a string argument (#16175)

The [original rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md) has a config option in the format of `["error", "element"]` or `["error", "syntax"]`. We required a config object like `["error", { "mode": "element" }]`, which is technically incompatible for existing configs that get migrated to oxlint.

This fixes that in the generated docs, and also makes it work correctly in the oxlint config. We retain compatibility with the object syntax in case anyone has set up oxlint with it already, and I've added tests to ensure both formats of the config option work.

Generated docs now:

```md
## Configuration

This rule accepts one of the following string values:

### `"syntax"`

This is the default mode. It will enforce the shorthand syntax for React fragments, with one exception.
Keys or attributes are not supported by the shorthand syntax, so the rule will not warn on standard-form fragments that use those.

Examples of **incorrect** code for this rule:
\```jsx
<React.Fragment><Foo /></React.Fragment>
\```

Examples of **correct** code for this rule:
\```jsx
<><Foo /></>
\```

\```jsx
<React.Fragment key="key"><Foo /></React.Fragment>
\```

### `"element"`

This mode enforces the standard form for React fragments.

Examples of **incorrect** code for this rule:
\```jsx
<><Foo /></>
\```

Examples of **correct** code for this rule:
\```jsx
<React.Fragment><Foo /></React.Fragment>
\```

\```jsx
<React.Fragment key="key"><Foo /></React.Fragment>
\```
```
